### PR TITLE
Add request payload support to the ballerina-to-openapi command

### LINKED ISSUE #28: Refactor the OpenAPI-CLI using new projectAPI
**Description:**
Need to re-solve the all the deprecated file usage with project API changes. Main purpose is to refactor the ballerina-to-openapi command. 
- [x] Create a project instance and traves it using syntax tree 
- [x] Refator the ballerina to openapi command implementation 
  - [x]  Paths without path parameters
  - [x] Paths with path parameters
  - [x] Support for query parameters

- **Request payload**
     - [x] Simple types
     - [x] Simple records
     - [x] Nested records 
     - [x] Nested record with nested array fields

**Suggested Labels:**

**Suggested Assignees:**

**Affected Product Version:**

**OS, DB, other environment details and versions:**    

**Steps to reproduce:**

**Related Issues:**

### PR DESCRIPTION
**Pull request by lnash94**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: openapi-cli/src/main/java/org/ballerinalang/ballerina/service/OpenApiServiceMapper.java [hunk 1]
```diff
@@ -47,17 +48,21 @@
...
     /**
      * Initializes a service parser for OpenApi.
-     *
-     * @param httpAlias    The alias for ballerina/http module.
+     *  @param httpAlias    The alias for ballerina/http module.
      * @param openApiAlias The alias for ballerina.openapi module.
+     * @param semanticModel
      */
-    public OpenApiServiceMapper(String httpAlias, String openApiAlias) {
+    public OpenApiServiceMapper(String httpAlias, String openApiAlias,
```
Comment (hemikak) [inline openapi-cli/src/main/java/org/ballerinalang/ballerina/service/OpenApiServiceMapper.java:60]: Can these be in the same line ?

File: openapi-cli/src/main/java/org/ballerinalang/ballerina/service/OpenApiServiceMapper.java [hunk 2]
```diff
@@ -47,17 +48,21 @@
...
+    private final SemanticModel semanticModel;
+
 
     /**
      * Initializes a service parser for OpenApi.
-     *
-     * @param httpAlias    The alias for ballerina/http module.
+     *  @param httpAlias    The alias for ballerina/http module.
      * @param openApiAlias The alias for ballerina.openapi module.
+     * @param semanticModel
```
Comment (hemikak) [inline openapi-cli/src/main/java/org/ballerinalang/ballerina/service/OpenApiServiceMapper.java:58]: Shall we fill in the documentation ?

File: openapi-cli/src/test/resources/ballerina-to-openapi/json_payload_service.bal [hunk 3]
```diff
@@ -0,0 +1,18 @@
+ import ballerina/http;
+ import ballerina/log;
+
+ listener http:Listener helloEp = new (9090);
+
+ service /payloadV on helloEp {
+
+     //resource function post hi(http:Caller caller, http:Request request, int id) {
+     //
+     //}
```
Comment (hemikak) [inline openapi-cli/src/test/resources/ballerina-to-openapi/json_payload_service.bal:10]: Are we going to uncomment these later ?

File: openapi-cli/src/test/resources/ballerina-to-openapi/json_payload_service.bal [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+     resource function post hi(http:Caller caller, http:Request request, @http:Payload {} json payload) {
+
+     }
+     //resource function post hi02(http:Caller caller, http:Request request, @http:Payload xml payload) {
+     //
+     //}
```
Comment (hemikak) [inline openapi-cli/src/test/resources/ballerina-to-openapi/json_payload_service.bal:16]: Are we going to uncomment these later ?

File: openapi-cli/src/test/resources/ballerina-to-openapi/project_bal/Ballerina.toml [hunk 5]
```diff
@@ -0,0 +1,4 @@
+[package]
+org= "hansani"
+name= "openAPI"
+version= "0.1.0"
```
Comment (hemikak) [inline openapi-cli/src/test/resources/ballerina-to-openapi/project_bal/Ballerina.toml:4]: Add a new line at the end.

File: openapi-cli/src/main/java/org/ballerinalang/ballerina/service/OpenApiResourceMapper.java [hunk 6]
```diff
@@ -249,6 +478,8 @@ private void addResourceParameters(FunctionDefinitionNode resource, OperationAda
      */
     private void createParametersModel(FunctionDefinitionNode resource, Operation operation) {
         List<Parameter> parameters = new LinkedList<>();
+        String in;
```
Comment (hemikak) [inline openapi-cli/src/main/java/org/ballerinalang/ballerina/service/OpenApiResourceMapper.java:481]: Is this variable used ?

File: openapi-cli/src/main/java/org/ballerinalang/ballerina/service/OpenApiResourceMapper.java [hunk 7]
```diff
@@ -213,34 +246,230 @@ private void addResourceParameters(FunctionDefinitionNode resource, OperationAda
...
+        if (arrayCount > 1) {
+            ArrayProperty input = new ArrayProperty();
+            input.setItems(new ArrayProperty());
+            property.setItems(handleArray(arrayCount - 1, symbolProperty, input));
+        } else {
+            property.setItems(symbolProperty);
+        }
+    }
+
+    private ArrayProperty handleArray(int count, Property property, ArrayProperty arrayProperty) {
```
Comment (hemikak) [inline openapi-cli/src/main/java/org/ballerinalang/ballerina/service/OpenApiResourceMapper.java:456]: Rename `count` as `arrayDimensions`.

File: openapi-cli/src/main/java/org/ballerinalang/ballerina/service/OpenApiResourceMapper.java [hunk 8]
```diff
@@ -213,34 +246,230 @@ private void addResourceParameters(FunctionDefinitionNode resource, OperationAda
...
+                TypeReferenceTypeSymbol typeRecord =
+                        (TypeReferenceTypeSymbol) arrayTypeSymbol.memberTypeDescriptor();
+                handleRecordPayload(queryParam, definitions, typeRecord);
+            }
+        }
+        //Handle nested array type
+        if (arrayCount > 1) {
+            ArrayProperty input = new ArrayProperty();
+            input.setItems(new ArrayProperty());
+            property.setItems(handleArray(arrayCount - 1, symbolProperty, input));
```
Comment (hemikak) [inline openapi-cli/src/main/java/org/ballerinalang/ballerina/service/OpenApiResourceMapper.java:450]: Can we try to move the creation of the first array inside the function as well ?

File: openapi-cli/src/main/java/org/ballerinalang/ballerina/service/OpenApiResourceMapper.java [hunk 9]
```diff
@@ -213,34 +246,230 @@ private void addResourceParameters(FunctionDefinitionNode resource, OperationAda
...
+            definitions.put(componentName, messageModel);
+            this.openApiDefinition.setDefinitions(definitions);
+        }
+    }
+
+    private void setArrayProperty(RequiredParameterNode queryParam, Map<String, Model> definitions, FieldSymbol field,
+                                  ArrayProperty property) {
+
+        TypeSymbol symbol = field.typeDescriptor();
+        int arrayCount = 0;
```
Comment (hemikak) [inline openapi-cli/src/main/java/org/ballerinalang/ballerina/service/OpenApiResourceMapper.java:421]: Rename as `arrayDimensions`

File: openapi-cli/src/main/java/org/ballerinalang/ballerina/service/OpenApiServiceMapper.java [hunk 10]
```diff
@@ -47,17 +48,21 @@
...
      */
-    public OpenApiServiceMapper(String httpAlias, String openApiAlias) {
+    public OpenApiServiceMapper(String httpAlias, String openApiAlias,
+                                SemanticModel semanticModel) {
```
Comment (hemikak) [inline openapi-cli/src/main/java/org/ballerinalang/ballerina/service/OpenApiServiceMapper.java:61]: Can these be in the same line ?

File: openapi-cli/src/test/resources/ballerina-to-openapi/json_payload_service.bal (hunk 3 above)
Comment (lnash94) [reply openapi-cli/src/test/resources/ballerina-to-openapi/json_payload_service.bal:10]: we can remove this comments , I will remove

File: openapi-cli/src/main/java/org/ballerinalang/ballerina/service/OpenApiResourceMapper.java (hunk 6 above)
Comment (lnash94) [reply openapi-cli/src/main/java/org/ballerinalang/ballerina/service/OpenApiResourceMapper.java:481]: No, we can remove this.

**Review by hemikak — APPROVED** [review]

**Pull request merged**